### PR TITLE
feat: import users from plex & fix: user delete button re-enable

### LINKED ISSUE #281: Import all users from Plex
**Is your feature request related to a problem? Please describe.**
Currently, once a user logs in then you can change their permissions. I believe a better workflow would be to import all the users and be able to set permissions ahead of time. 

**Describe the solution you'd like**
Be able to manage users before they log in. I believe a sync job to Plex to collect all the users will be needed and should be ran periodically.

### PR DESCRIPTION
**Pull request by johnpyp**
#### Description

Adds an api endpoint and corresponding UI button for importing users from the connected Plex server, as initially proposed in #281. 

Specifically, it imports all "shared library users" that are real users (have a corresponding username and email). Most of the user properties are imported, except notably the `plexToken`, but the auth flow takes care of populating that after first-login.

This PR also includes a minor one-line fix for the delete user button in the popup modal not re-enabling after you've already deleted one user, though I can move it into a different PR if desired.

#### Screenshot (if UI related)

Import Button

#### Todos

- [x] Sucessfully builds `yarn build`
- [x] Translation Keys `yarn i18n:extract`
- [ ] Database migration created (if required)

#### Issues Fixed or Closed by this PR

- Fixes #281

### CODE REVIEW

File: src/components/UserList/index.tsx [hunk 1]
```diff
@@ -66,10 +70,31 @@ const UserList: React.FC = () => {
...
 
+  const importFromPlex = async () => {
+    setImporting(true);
+
+    try {
+      await axios.post('/api/v1/user/import-from-plex');
+      addToast(intl.formatMessage(messages.importedfromplex), {
+        autoDismiss: true,
+        appearance: 'success',
+      });
```
Comment (sct) [inline src/components/UserList/index.tsx:86]: What about having the toast say the # of new users that were added? `4 new users imported from Plex` for example.

File: src/components/UserList/index.tsx [hunk 2]
```diff
@@ -66,10 +70,31 @@ const UserList: React.FC = () => {
         appearance: 'error',
       });
     } finally {
+      setDeleting(false);
```
Comment (sct) [inline src/components/UserList/index.tsx:73]: whoops. thanks!

File: src/components/UserList/index.tsx [hunk 3]
```diff
@@ -18,6 +18,9 @@ import globalMessages from '../../i18n/globalMessages';
 
 const messages = defineMessages({
   userlist: 'User List',
+  importfromplex: 'Import Users From Plex',
+  importfromplexerror: 'Something went wrong importing users from plex',
+  importedfromplex: 'Users successfully imported from plex',
```
Comment (sct) [inline src/components/UserList/index.tsx:23]: Plex should be capitalized here.

File: server/routes/user.ts [hunk 4]
```diff
@@ -142,4 +143,51 @@ router.delete<{ id: string }>('/:id', async (req, res, next) => {
   }
 });
 
+router.post('/import-from-plex', async (req, res, next) => {
+  try {
+    const defaultPermission = req.body.permissions || Permission.REQUEST;
```
Comment (sct) [inline server/routes/user.ts:148]: We can now set default permissions through the settings page so you can grab the current `defaultPermissions` value from the settings here.

**Review by sct — COMMENTED** [review]
Fantastic PR! Looks really good!

Just a few small changes and we can get this merged!

**Comment by sct** [comment]
Also don't forget to run `yarn i18n:extract` and commit the generated `en.json` file!

**Comment by johnpyp** [comment]
Alright, these changes should be in there now 👍

File: src/components/UserList/index.tsx [hunk 5]
```diff
@@ -66,10 +70,38 @@ const UserList: React.FC = () => {
...
+  const importFromPlex = async () => {
+    setImporting(true);
+
+    try {
+      const { data: createdUsers } = await axios.post(
+        '/api/v1/user/import-from-plex'
+      );
+      addToast(
+        `${createdUsers.length} ${intl.formatMessage(
+          messages.importedfromplex
```
Comment (sct) [inline src/components/UserList/index.tsx:87]: Not every language will put the number first. The actual localized string needs to use the number in it. `{userCount} new users imported from Plex`. Then you pass it the value of `userCount` as the second argument of formatMessage in an object.

Comment (johnpyp) [reply src/components/UserList/index.tsx]: Ah gotcha. I took a look at the docs for react-intl and went with the 3-branches strategy of "no new users", "1 new user", and "N new users".

**Comment by sct** [comment]
I think it's safe to take this out of draft 😁

**Comment by johnpyp** [comment]
ah yep 👍

**Review by sct — APPROVED** [review]
Perfect. Thanks a ton!

**Comment by sct** [comment]
@all-contributors please add @johnpyp for code

**Pull request merged**